rotkehlchen/db/upgrades/v43_v44.py

```diff
@@ -62,18 +62,64 @@ def _remove_log_removed_column(write_cursor: 'DBCursor') -> None:
     )
 
 
+def _upgrade_account_tags(write_cursor: 'DBCursor') -> None:
+    """Upgrade the object_reference references in tag_mappings to not
+    depend on the supported blockchain
+    """
+    object_references: list[tuple[str, str]] = write_cursor.execute(
+        'SELECT object_reference, tag_name FROM tag_mappings',
+    ).fetchall()
+    if len(object_references) == 0:
+        return
+
+    remove_keys, insert_entries = [], set()
+    suppported_blockchains = (  # supported blockchains at the moment of writing the upgrade
+        'ETH',
+        'ETH2',
+        'BTC',
+        'BCH',
+        'KSM',
+        'AVAX',
+        'DOT',
+        'OPTIMISM',
+        'POLYGON_POS',
+        'ARBITRUM_ONE',
+        'BASE',
+        'GNOSIS',
+        'SCROLL',
+        'ZKSYNC_LITE',
+    )
+    for (object_reference, tag_name) in object_references:
+        for supported_chain in suppported_blockchains:
+            if object_reference.startswith(supported_chain):
+                remove_keys.append(object_reference)
+                insert_entries.add((object_reference.removeprefix(supported_chain), tag_name))
+                break
```

Comment:
we can use a regex here like: `^(ETH|ETH2|BTC|BCH|KSM|AVAX|DOT|OPTIMISM|POLYGON_POS|ARBITRUM_ONE|BASE|GNOSIS|SCROLL|ZKSYNC_LITE)`
and match it with the `object_reference`, then use `re.sub` to remove that part. This way we can avoid running a nested loop for each entry.